Add support for templates with mixed coalitions

ricmzn opened the pull request:
This allows a template to contain groups of more than one, on the condition that it is specified in the template's .dct. This can be useful for things such as adding friendly JTACs that are linked to a specific conflict, TACAN navigation beacons, or even friendly troops that can be transported to an objective (via the game's built-in unit transport system).

If the template doesn't have any targets specified, goals will only be auto-generated for objects that belong to the same coalition as the template, but the mission designer is free to explicitly mark neutral or even friendly units as targets (if they find an use case for it).

This also fixes #135 by removing the type requirement from the `coalition` key in templates. The validation method still checks that the input is valid, either in number or string form, or continues if nil (so that auto-detection may be applied).

A new template has been added to the demo theater to test validation of this new feature automatically.

ricmzn commented on src/dct/templates/Template.lua:
```diff
@@ -145,8 +146,7 @@ local function checkbldgdata(keydata, tpl)
 		local bldgdata = {}
-		bldgdata.countryid = 0
-		bldgdata.category  = enum.UNIT_CAT_SCENERY
+		bldgdata.category = enum.UNIT_CAT_SCENERY
```
Scenery doesn't have a country associated anymore, so coalitions cannot be detected from it anymore. This is intentional, as countryid 0 would auto-assign it to Russia.

jtoppins commented on src/dct/templates/STM.lua:
```diff
@@ -122,7 +122,7 @@ end
 --    }}}
 --]]
 
-function STM.transform(stmdata, file)
```
I think you still need to keep the `file` function parameter or edit src/dct/templates/Template.lua +520 to not pass the file name. The file name is provided by the utils.readlua call. Also, it would seem useful to keep the filename name for error reporting which is what originally it was used for.

ricmzn replied:
As the `file` parameter is not used here anymore (the coalition check was moved to the Template validation), one alternative could be to set it as a field of Template.

jtoppins replied:
Ok my only point was where `STM.transform` is used when called with `readlua`, `STM.transform(utils.readlua(...))`, readlua returns a two tuple of stmdata and the file path.

jtoppins commented:
@ricmzn I re-targeted this PR to the next branch. If you could update the PR so that it merges onto this branch that would help.

ricmzn commented:
Rebased the branch and addressed the unused `file` parameter in `STM.transform`.

I'd also like to normalize the template path emitted in errors but there's a non-trivial amount of complexity in handling both Windows and Linux paths.

jtoppins commented:
> I'd also like to normalize the template path emitted in errors but there's a non-trivial amount of complexity in handling both Windows and Linux paths (within Lua's limited string handling).

Since we require all DCT related data to be relative to `lfs.writedir()`, does knowing this make it easier for normalization purposes? Are you concerned with log output or general receive some arbitrary string and go looking it up in a file system?

ricmzn commented:
That's a good point. I don't think parent folder segments are used anywhere in DCT outside of tests?

jtoppins commented:
I think that is correct.
